### PR TITLE
Add database deletion API functions

### LINKED ISSUE #31: Ensure Database deletion API coverage similar to Android
We support the following Database deletion methods on Android:

```
AzureData.deleteDatabase (id) {
    // successfully deleted == it.isSuccessful
}

AzureData.deleteDatabase (database) {
    // successfully deleted == it.isSuccessful
}

database.delete {
    // successfully deleted == it.isSuccessful
}
```

Need to:

- [ ] Make sure all APIs/overloads are supported in iOS library
- [ ] Add examples to the [docs](https://github.com/Azure/Azure.iOS/wiki/AzureData)

### PR DESCRIPTION
**Pull request by ftchirou**
Fixes #31 .

Changes Proposed in this pull request:
- Add database deletion API functions
- Update the docs

### CODE REVIEW

File: AzureData/Source/AzureData.swift [hunk 1]
```diff
@@ -79,9 +79,13 @@ public func get (databaseWithId databaseId: String, callback: @escaping (Respons
 }
 
 /// Delete a Database
-//public func delete (_ database: Database, callback: @escaping (DataResponse) -> ()) {
-//    return DocumentClient.default.delete (database, callback: callback)
-//}
+public func delete (_ database: Database, callback: @escaping (DataResponse) -> ()) {
```
Comment (colbylwilliams) [inline AzureData/Source/AzureData.swift]: I don't believe this override is needed as it's already handled by the generic func here:
https://github.com/Azure/Azure.iOS/blob/31d13a1306d11fe35fb23730931318c67fe0d3f5/AzureData/Source/AzureData.swift#L484-L486

File: AzureData/Source/Resources/ResourceOracle.swift [hunk 2]
```diff
@@ -98,15 +98,43 @@ public class ResourceOracle {
...
         }
     }
-    
+
+    fileprivate static func _removeLinks(forResourceWithLink link: String) {
+
+        if let selfLink = selfLinkLookup[link] {
+            selfLinkLookup[link] = nil
+
+            if let _ = altLinkLookup[selfLink] {
```
Comment (colbylwilliams) [inline AzureData/Source/Resources/ResourceOracle.swift]: I don't think it's necessary to check for the existence of the key before attempting to remove it via subscript.

File: AzureData/Source/Resources/ResourceOracle.swift [hunk 3]
```diff
@@ -98,15 +98,43 @@ public class ResourceOracle {
...
+                altLinkLookup[selfLink] = nil
+            }
+
+            return
+        }
+
+        if let altLink = altLinkLookup[link] {
+            altLinkLookup[link] = nil
+
+            if let _ = selfLinkLookup[altLink] {
```
Comment (colbylwilliams) [inline AzureData/Source/Resources/ResourceOracle.swift]: I don't think it's necessary to check for the existence of the key before attempting to remove it via subscript

File: AzureData/Source/AzureData.swift (hunk 1 above)
Comment (ftchirou) [reply AzureData/Source/AzureData.swift]: Oh! That's right. Thanks for catching that.

**Comment by ftchirou** [comment]
I addressed your comments. @colbylwilliams

**Comment by colbylwilliams** [comment]
merged and deleted branch

**Pull request merged**